packages/code/src/CopyButton/CopyButton.tsx

```diff
@@ -51,19 +56,45 @@ function CopyButton({ onCopy, contents }: CopyProps) {
     setCopied(true);
   };
 
+  const handleMouseEnter = () => {
+    setHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setHovered(false);
+  };
+
+  const shouldClose = () => !hovered;
+
   return (
-    <IconButton
-      ref={setButtonNode}
+    <Tooltip
+      open={hovered}
+      setOpen={setHovered}
       darkMode={darkMode}
-      aria-label="Copy"
-      className={cx(copyButtonThemeStyles[theme], {
-        [copiedThemeStyle[theme]]: copied,
-      })}
-      onClick={handleClick}
+      align={Align.Top}
+      justify={Justify.Middle}
+      trigger={
+        <IconButton
+          ref={setButtonNode}
+          darkMode={darkMode}
+          aria-label={COPY_TEXT}
+          className={cx(copyButtonThemeStyles[theme], {
+            [copiedThemeStyle[theme]]: copied,
+          })}
+          onClick={handleClick}
+          onMouseEnter={handleMouseEnter}
```

Comment:
Would this still work if you removed `onMouseEnter` and `onMouseLeave`? The tooltip should open on hover by default.

Reply:
https://github.com/mongodb/leafygreen-ui/assets/52300529/77788845-4031-4268-97f9-80c305fcfa2e

yeah initially tried to implement with the default api behavior, but I ran into issues with component re-renders because `CopyButton` re-renders when the `copied` state toggles on/off. That resulted in the above inconsistent UX. By setting up state to track the `hovered` state and making the tooltip `open` / `setOpen` controlled by `hovered`, I was able to keep the `Tooltip` open regardless of `CopyButton` re-rendering

Reply:
I see. Don't forget to add `onFocus` and `onBlur`. Right now, the tooltip remains open if you tab away from the copy button.

Reply:
https://github.com/mongodb/leafygreen-ui/assets/52300529/c3188850-a59e-4854-b664-d48bf22dedf8
I ended up needing to use `handleKeyDown` instead of `onFocus`/`onBlur` because using `onBlur` conflicts with the mouse event handlers I have in place. I have it specifically close out for tab. Nothing else in the `keyMap` seemed to require action although I did consider `Escape` key

@shaneeza would appreciate your feedback on this so I can stay consistent with the codebase. Thanks!

Reply:
`handleKeyDown` doesn't feel correct, how was `onBlur` conflicting with the mouse event handlers?

Reply:
Also, should the tooltip still appear on hover after clicking on it? It seems like the tooltip would still be open but the text would change back to `copy`

Reply:
> how was `onBlur` conflicting with the mouse event handlers?
I set up `hovered` state to control the open/close of the tooltip. If I use `onBlur` to toggle off `hovered`, this will happen:
1. mouse enter: hovered set to true and tooltip opens
2. click: copied set to true which makes `trigger` button re-render which causes closed tooltip to re-render
3. mouse enter on newly rendered button: hovered set to true and tooltip re-opens

> Also, should the tooltip still appear on hover after clicking on it? It seems like the tooltip would still be open but the text would change back to copy
ahh yes, the change I made for keyboard interaction was causing it to close, but I just pushed a change that should support both cases now. will update the videos in the PR description